Docs and CI

Kiraprint opened the pull request:
Added gh workflow with python 3.10 build for pushes check.
Added base documentation for project (need to edit some lines by developers and write a Quick Start)

Kiraprint committed:
Created workflow.

Create .github/workflows/python-package.yml

Kiraprint committed:
Update attention.py

Edited missed torch functional module import.

Kiraprint committed:
Update python-package.yml

Kiraprint committed:
Update python-package.yml

Kiraprint committed:
Update python-package.yml

Kiraprint committed:
Update python-package.yml

Kiraprint committed:
Created requirements.txt

Kiraprint committed:
pep8 bot configured

Kiraprint committed:
Update python-package.yml

Kiraprint committed:
Update python-package.yml

Kiraprint committed:
Merge pull request #2 from Kiraprint/Kiraprint-patch-1

Update python-package.yml

Kiraprint committed:
added documentation base

Kiraprint committed:
Solved logo problem in readme

Same as previous

Kiraprint committed:
Solved logo problem in readme

Kiraprint committed:
Same as previous

Kiraprint committed:
Update requirements.txt

Kiraprint committed:
Update requirements.txt

Kiraprint committed:
Merge branch 'main' of https://github.com/Kiraprint/envintegration

Kiraprint committed:
Merge branch 'Kiraprint-patch-1' of https://github.com/Kiraprint/envintegration into Kiraprint-patch-1

Kiraprint committed:
Merge pull request #3 from Kiraprint/main

Merging docs and bots changes

Kiraprint committed:
modified:   .github/workflows/python-package.yml
	deleted:    requirements.txt

Kiraprint committed:
Merge pull request #4 from Kiraprint/Kiraprint-patch-1

Documentation and CI

AndrewLaptev commented on .github/workflows/python-package.yml:
```diff
@@ -0,0 +1,41 @@
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10"]
```
А зачем тут 3.8, 3.9?

Kiraprint replied:
Исправил, т.к. действительно незачем.

AndrewLaptev commented on .github/workflows/python-package.yml:
```diff
@@ -0,0 +1,41 @@
+        python -m pip install --upgrade pip
+        python -m pip install flake8 pytest pytest-cov
+        pip install pip-tools
+        python -m piptools compile -o requirements.txt pyproject.toml
```
Зачем здесь файл с рикварами? Я же говорил про то, что pip зависимости уже давно умеент через pyproject тянуть. Тем более requirements.txt нет в репе.

Kiraprint replied:
Ок, не знал, что это делается через
pip install .
Исправил
Но вообще эта строчка внутри workflow компилирует requirements.txt на основе pyproject.toml, поэтому она сработает вне зависимости от наличия первого.

AndrewLaptev commented on .pep8speaks.yml:
```diff
@@ -0,0 +1,28 @@
+# File : .pep8speaks.yml
```
А зачем вообще этот файлик?

Kiraprint replied:
https://github.com/ITMO-NSS-team/open-source-ops/blob/master/tutorials/setup_bots.rst
Для бота PEP8. Нужно попробовать добавить в наш проект, но на моём форке он почему-то не сработал.

AndrewLaptev replied:
Понял, спс. Можно ли этому боту указать правила с линтеров, которые мы уже настроили в pyproject? Этот бот обязателен вообще?

Kiraprint replied:
Можно как раз в этот файл записать те же значения, что и в pyproject для линтеров. Насчёт обязательности - скорее да. Стараюсь опираться на наши сформированные [гайды](https://github.com/ITMO-NSS-team/open-source-ops/blob/master/tutorials/quick_guide.md) по ведению репо

AndrewLaptev replied:
Окей. Тогда перенеси, пожалуйста, необходимые конфиги для линтеров и форматтеров из pyproject и .flake8 файлов в этот .pep8speaks.yml файлик

Kiraprint committed:
On branch Kiraprint-patch-1
 Your branch is up to date with 'origin/Kiraprint-patch-1'.

 Changes to be committed:
	modified:   .github/workflows/python-package.yml

Kiraprint committed:
Merge pull request #5 from Kiraprint/Kiraprint-patch-1

On branch Kiraprint-patch-1

Kiraprint committed:
Update README.md

Fixed 1st table's view

Kiraprint committed:
Added pyproject docs dependency group

Update .readthedocs.yaml

Kiraprint committed:
Update python-package.yml

Added pre-commit hooks check.

Kiraprint committed:
Added pyproject docs dependency group

Kiraprint committed:
Added pyproject docs dependency group

Kiraprint committed:
Edited readthedocs.yaml

Kiraprint committed:
Edited toml for docs

Kiraprint committed:
Edited toml for docs

Kiraprint committed:
Edited toml for docs

Kiraprint committed:
Edited toml for docs

Kiraprint committed:
Edited toml for docs

Kiraprint committed:
Edited toml for docs

Kiraprint committed:
Update .readthedocs.yaml

Kiraprint committed:
Update pyproject.toml

Kiraprint committed:
Update .readthedocs.yaml

Kiraprint committed:
Update .readthedocs.yaml

Kiraprint committed:
Merge branch 'Kiraprint-patch-1' of https://github.com/Kiraprint/envintegration into Kiraprint-patch-1

Kiraprint committed:
Edited toml docs

Kiraprint committed:
Update .pre-commit-config.yaml

Kiraprint committed:
Update .pre-commit-config.yaml

Kiraprint committed:
Update .pre-commit-config.yaml

Kiraprint committed:
Update .pre-commit-config.yaml

Kiraprint committed:
Update .pre-commit-config.yaml

Kiraprint committed:
Update .pre-commit-config.yaml

Kiraprint committed:
Update .pre-commit-config.yaml

Kiraprint committed:
Update .pre-commit-config.yaml

Kiraprint committed:
Update python-package.yml

Kiraprint committed:
Update python-package.yml

Kiraprint committed:
Update python-package.yml

Kiraprint committed:
Update python-package.yml

Kiraprint committed:
Update .pre-commit-config.yaml

Kiraprint committed:
Update .pep8speaks.yml

Kiraprint committed:
Edited pre-commit.yml

Kiraprint committed:
Added some tests for TextEmotionApi

Kiraprint committed:
Added some tests for TextEmotionApi

Kiraprint committed:
Merge pull request #6 from Kiraprint/Kiraprint-patch-1

Docs and CI
